docs/guides/src/main/server/enabletls.adoc

```diff
@@ -0,0 +1,70 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/kc.adoc" as kc>
+
+<@tmpl.guide
+title="Configure Keycloak to use TLS"
+summary="Learn how to use HTTPS/TLS with Keycloak">
+
+Transport Layer Security (short: TLS) is crucial to exchange data over a secured channel. For production environments, you should never expose Keycloak endpoints through HTTP, as sensitive data is at the core of what Keycloak exchanges with other applications. In this guide you will learn how to configure Keycloak to use HTTPS/TLS.
+
+== Configure TLS in Keycloak
+Keycloak can be configured to load the needed certificate infrastructure either from a certificate / key file pair in PEM format or from an existing keystore file. When both alternatives are set up, the certificate / key file configuration takes precedence over the keystore configuration.
+
+=== Providing a certificate / key file pair
+When you use a pair of matching certificate / private key files in PEM format, configure Keycloak to use them by running:
+
+<@kc.all parameters="--https-certificate-file=/path/to/certfile.pem --https-certificate-key-file=/path/to/keyfile.pem"/>
+
+Keycloak creates a keystore out of these files in memory and uses this keystore afterwards.
+
+=== Providing a keystore
+As an alternative, you can use an existing keystore by running:
+<@kc.all parameters="--https-key-store-file=/path/to/existing-keystore-file"/>
+
+When no keystore file is configured explicitly, but http.enabled is set to false, Keycloak will try to use a `server.keystore` file in the `conf` subdirectory by default.
+
+[NOTE]
+No matter which alternative you choose for setting up the TLS configuration, please make sure to set a secure password for your keystore using the `https.key-store.password` option. We recommend to not set the password in plaintext via CLI or in `conf/keycloak.properties`, but instead using good practices such as using a vault or an environment variable - both can be substituted in the properties file. If no password is set, the default password `password` is used.
```

Comment:
Why do we need to set a password when using PEM files?

Reply:
that was a guess, but I got an answer from the quarkus team here: https://github.com/quarkusio/quarkus/discussions/22851#discussioncomment-1961299 so yes, vert.x stores it internally in a keystore in memory, so the question is now: should we consider this "safe enough" to not advocate for using a safe password? Fine either way, tbh :)

Reply:
I do not think there is two ways, but no need to set a password if using PEM files.

Reply:
Updated, can you pls check again?